server/monitor-types/tls.js

```diff
@@ -0,0 +1,266 @@
+                servername: hostname
+            });
+            this.debug_log("TLS connected");
+            return tlsSocket;
+        }
+    }
+
+    /**
+     * Reads available data from the given socket.
+     * @param {AbortController} aborter Abort controller used to abort the read
+     * @param {*}               socket  net.Socket or tls.TLSSocket instance to use
+     * @returns {Promise<string>} Data read from the socket or rejected promise on error
+     */
+    readData(aborter, socket) {
+        return new Promise((resolve, reject) => {
+            const cleanup = function () {
+                socket.removeListener("error", onError);
+                socket.removeListener("data", onData);
+                socket.pause();
+                aborter.removeEventListener("abort", onAbort);
+            };
+
+            const onAbort = (_) => {
+                this.debug_log("read aborted");
+                cleanup();
+                reject(new Error("Timeout"));
+            };
+
+            const onError = (error) => {
+                this.debug_log(`unable to read data: ${error}`);
+                cleanup();
+                reject(new Error(`Failed to read data: ${error}`));
+            };
+
+            const onData = (data) => {
+                const dataString = data.toString().trim();
+                this.debug_log(`read data: '${dataString}'`);
+                cleanup();
+                resolve(dataString);
+            };
+
+            aborter.addEventListener("abort", onAbort, { once: true });
+
+            socket.on("error", onError);
+            socket.on("data", onData);
+            socket.resume();
+        });
+    }
+
+    /**
+     * Reads available data from the given socket if it starts with a given prefix.
+     * @param {AbortController} aborter  Abort controller used to abort the read
+     * @param {*}               socket   net.Socket or tls.TLSSocket instance to use
+     * @param {string}          expected Prefix the response is expected to start with
+     * @returns {Promise<string>} Data read from the socket or rejected promise if the response does
+     *                            not start with the prefix
+     */
+    async expectDataStartsWith(aborter, socket, expected) {
+        const data = await this.readData(aborter, socket);
+        this.debug_log(`response data: '${data}', expected: '${expected}'…`);
+        if (!data.startsWith(expected)) {
+            throw new Error(`Unexpected response. Data: '${data}', Expected: '${expected}'…`);
+        }
+    }
+
+    /**
+     * Performs STARTTLS on the given socket.
+     * @param {AbortController} aborter Abort controller used to abort the STARTTLS process
+     * @param {*}               socket  net.Socket or tls.TLSSocket instance to use
+     * @returns {Promise<void>} Rejected promise if the STARTTLS process failed
+     */
+    async startTls(aborter, socket) {
+        this.debug_log("waiting for prompt");
```

Comment:
lets make this more apparent to other readers ^^

```suggestion
        // 220 in the context of RFC3207 means "Ready to start TLS"
        this.debug_log("waiting for prompt");
```

Reply:
I'll actually have to check if my "220" assumption is correct for all (common) protocols that support STARTTLS. :thinking:

Reply:
As I mentioned elsewhere, this has now changed since all those strings have become parametrized and can be changed in the UI.